aist2019-openvino-semseg/semseg_inference.py

```diff
@@ -1,7 +1,143 @@
+        device to infer on; CPU, GPU, FPGA or MYRIAD is acceptable. \
+        Sample will look for a suitable plugin for device specified \
+        (CPU by default)', default='CPU', type=str)
+    parser.add_argument('--color_map', help='Classes color map', type=str,
+        default=None)
+    return parser
+
+def prepare_model(model, weights, cpu_extension, device, plugin_dirs, log):
+    model_xml = model
+    model_bin = weights
+
+    log.info('Creating plugin object for inference')
+    plugin = IEPlugin(device=device, plugin_dirs=plugin_dirs)
+    if cpu_extension and 'CPU' in device:	
+        plugin.add_cpu_extension(args.cpu_extension)
+
+    log.info('Loading network files:\n\t{}\n\t{}'.format(model_xml, model_bin))
+    net = IENetwork(model=model_xml, weights=model_bin)
+
+    log.info('Checking layers are supported by plugin')
+    if 'CPU' in plugin.device:
+        supported_layers = plugin.get_supported_layers(net)
+        not_supported_layers = [l for l in net.layers.keys() \
+            if l not in supported_layers]
+        if len(not_supported_layers) != 0:
+            log.error('Following layers are not supported by the plugin for specified' 
+                'device {}:\n {}'.format(plugin.device, ', '.join(not_supported_layers)))
+            log.error('Please try to specify cpu extensions library path in sample\'s' 
+                'command line parameters using -l or --cpu_extension command line'
+                'argument')
+            sys.exit(1)
+    return net, plugin
+
+def get_image_list(input, log):
+    log.info('Creating list of input images')
+    if os.path.isdir(input[0]):
+        data = [os.path.join(input[0], file) for file in os.listdir(input[0])]
+    else:
+        data = input
+    return data
+
+def convert_image(net, data, log):
+    log.info('Getting shape of input tensor')
+    n, c, h, w = net.inputs[next(iter(net.inputs))].shape
+    log.info('Creating input images tensor')
+    images = np.ndarray(shape=(len(data), c, h, w))
+    for i in range(n):
+        image = cv2.imread(data[i])
+        if image.shape[:-1] != (h, w):
+            image = cv2.resize(image, (w, h))
+        image = image.transpose((2, 0, 1))
+        images[i] = image
+    return images
+
+def infer_sync(images, exec_net, net):
+    input_blob = next(iter(net.inputs))
+    out_blob = next(iter(net.outputs))
+    res = exec_net.infer(inputs={input_blob : images})
+    res = res[out_blob]
+    return res
+
+def segmentation_output(res, color_map, log):
+    c = 3
+    h, w = res.shape[2:]
+
+    log.info('Reading color map')
+    if not color_map:
+        color_map = 'color_map.txt'
+    classes_color_map=[]
+    with open(color_map, 'r') as f:
+        for line in f:
+            classes_color_map.append([int(x) for x in line.split()])
+
+    log.info('Creating segmented images')
+    for batch, data in enumerate(res):
+        classes_map = np.zeros(shape=(h, w, c), dtype=np.int)
+        for i in range(h):
```

Comment:
Maybe it's reasonable to use one level for-loop instead of nested:

```python
from itertools import product

for i, j in product(range(h), range(w)):
```